image_brightener.cpp

```diff
@@ -0,0 +1,37 @@
+#include "image_brightener.h"
+
+// For brightening, we add a certain grayscale (25) to every pixel.
+const int BRIGHTENING_VALUE = 25;
+const int MAX_PIXEL_VALUE = 255;
+
+ImageBrightener::ImageBrightener(Image& inputImage) : m_inputImage(inputImage) {
+}
+
+void ImageBrightener::BrightenWholeImage() {
+	uint32_t attenuatedPixelCount = 0;
+	for (uint32_t x = 0; x < m_inputImage.GetRows(); x++) {
+		for (uint32_t y = 0; y < m_inputImage.GetColumns(); y++) {
+			BrightenPixel(x, y);
+		}
+	}
+}
+
+void ImageBrightener::BrightenPixel(uint32_t x, uint32_t y) {
+	int pixelIndex = x * m_inputImage.GetColumns() + y;
```

Comment:
This is knowledge of the orientation of pixels - the fact that pixels are sequenced row by row.
Should this knowledge belong in the brightener? or in the image?

Reply:
agree . moved this logic to image class